chain/epoch_manager/src/shard_assignment.rs

```diff
@@ -0,0 +1,274 @@
+use near_primitives::types::validator_stake::ValidatorStake;
+use near_primitives::types::Balance;
+use near_primitives::utils::min_heap::MinHeap;
+use std::cmp;
+
+/// Assign chunk producers (a.k.a validators) to shards. The i-th element
+/// of the output corresponds to the validators assigned to the i-th shard.
+/// This function ensures that every shard has at least `min_validators_per_shard`
+/// assigned to it, and attempts to balance the stakes between shards (keep the total
+/// stake assigned to each shard approximately equal). This function performs
+/// best when the number of chunk producers is greater than
+/// `num_shards * min_validators_per_shard`.
+pub fn assign_shards<T: HasStake + Eq + Clone>(
+    chunk_producers: Vec<T>,
```

Comment:
Are `chunk_producers` sorted in any particular way (in particular, in descending order of stake)?
It would be good to document here.
If they are not sorted in descending stake, it might be worth doing?